src/components/Asset/AssetActions/Trade/index.tsx

```diff
@@ -30,7 +34,7 @@ export default function Trade(): ReactElement {
     async function getTokenBalance() {
       const datatokenInstance = new Datatoken(web3)
       const dtBalance = await datatokenInstance.balance(
-        asset.services[0].datatokenAddress,
+        asset.accessDetails.datatoken.address,
```

Comment:
just for consistency, throughout this PR, the datatoken info should always come from the same place as I found e.g. for address:

- `asset.services[0].datatokenAddress`
- `asset.accessDetails.datatoken.address`
- `poolInfo.datatokenAddress`

Considering that in future we might have multiple services and multiple datatokens, most future proof in this context here seems to be `poolInfo.datatokenAddress` as for swap we're interacting with a specific pool.

Reply:
`Considering that in future we might have multiple services and multiple datatokens` for this we need to change everything 🤣 .But consistency is good  nonetheless.